feat(content-gifting): custom article link expiration

miguelpeixe opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

NPPD-1011

Implements configurable expiration for the gifted article link.

<img width="1087" height="987" alt="image" src="https://github.com/user-attachments/assets/e77f59f7-46b6-498e-88d2-b5881f6e2859" />

The default is 5 days, and the modal text has been updated to fit the dynamic value better:

<img width="446" height="385" alt="image" src="https://github.com/user-attachments/assets/eeaa8152-85ec-437c-89b7-5c899b0d4259" />

### How to test the changes in this Pull Request:

1. Confirm the new unit test is accurate
2. Navigate to Audience -> Configuration -> Content Gate
3. Tweak the default settings to 1 hour
4. Refresh the page and confirm the saved options persist
5. As a subscriber, generate an article link
6. In an anonymous session, access the link and confirm it unlocks the content and you get the banner CTA
7. Confirm the created cookie has an expiration that matches the configured link expiration
8. Wait an hour, reaccess the link, and confirm it no longer unlocks the content

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

miguelpeixe committed:
feat(content-gifting): custom article link expiration

rbcorrales commented:
I haven't had the chance to fully review this functionality, but I noticed these things in the code:

1. There's a hardcoded 24-hour cookie expiration. This needs to use the configurable expiration time, otherwise users will lose access after 24 hours, even if you set a longer expiration. Or will keep access longer than intended for shorter than one-day expirations.
https://github.com/Automattic/newspack-plugin/blob/0a1e634d96758720f1fff9adaa3621c13a3b61b1/src/content-gate/content-gifting.js#L105

2. The description in the wizard still says "Allow members to gift articles for 24 hours"
https://github.com/Automattic/newspack-plugin/blob/0a1e634d96758720f1fff9adaa3621c13a3b61b1/src/wizards/audience/views/setup/content-gating.js#L89

miguelpeixe committed:
fix: wizard phrasing and cookie expiration

miguelpeixe commented:
Thank you for catching that! Fixed in 22ebf7c54a9359b97d1cbc4b1646b75605d1b582 and updated the PR description to include testing the cookie expiration.

rbcorrales reviewed (changes requested):
I'm testing this, trying to set the expiration to 1 hour, but the value is not being saved and set back to 5 days on save.
Here's the request payload:
```json
{
  "content_gifting": {
    "enabled": true,
    "limit": 10,
    "interval": "month",
    "expiration_time": 1,
    "expiration_time_unit": "hours",
    "cta_label": "This article has been gifted to you by me, someone who values great journalism.",
    "button_label": "Subscribe now",
    "cta_url": "",
    "style": "light"
  }
}
```

And the `content_gifting` object from the response:
```json
{
    "enabled": true,
    "limit": 10,
    "interval": "month",
    "expiration_time": 5,
    "expiration_time_unit": "days",
    "cta_label": "This article has been gifted to you by me, someone who values great journalism.",
    "button_label": "Subscribe now",
    "cta_url": "",
    "style": "light"
}
```

miguelpeixe committed:
fix: autoload false for gifting settings

rbcorrales reviewed (approved):
After flushing the object cache, the settings appear as expected.
Tested the flow and verified that the expiration date of the cookie matched what was configured.
